packages/hdwallet-provider/src/index.ts

```diff
@@ -36,7 +44,10 @@ class HDWalletProvider {
   private walletHdpath: string;
   private wallets: { [address: string]: ethJSWallet };
   private addresses: string[];
-  private chainId?: number | string;
+  private chainId?: number;
+  private chainSettings: any;
+  private hardfork: string;
+  private initialized: Promise<any>;
```

Comment:
Since this won't actually have a value, we can use `void`

```suggestion
  private initialized: Promise<void>;
```